bin/plugin/commands/translations.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+const glob = require( 'fast-glob' );
+const fs = require( 'fs' );
+const readline = require( 'readline' );
+
+/**
+ * Internal dependencies
+ */
+const { log, formats } = require( '../lib/logger' );
+const config = require( '../config' );
+
+/**
+ * @typedef WPTranslationsCommandOptions
+ *
+ * @property {string=} directory Optional directory, default is the root `/modules` directory.
+ * @property {string=} output    Optional output PHP file, default is the root `/module-i18n.php` file.
+ */
+
+/**
+ * @typedef WPTranslationsSettings
+ *
+ * @property {string} textDomain Plugin text domain.
+ * @property {string} directory  Modules directory.
+ * @property {string} output     Output PHP file.
+ */
+
+const options = [
+	{
+		argname: '-d, --directory <directory>',
+		description: 'Modules directory',
+	},
+	{
+		argname: '-d, --output <output>',
+		description: 'Output file',
+	},
+];
+
+/**
+ * Command that generates a PHP file from module header translation strings.
+ *
+ * @param {WPTranslationsCommandOptions} opt
+ */
+async function handler( opt ) {
+	await createTranslations( {
+		textDomain: config.textDomain,
+		directory: opt.directory || 'modules',
+		output: opt.output || 'module-i18n.php',
+	} );
+}
+
+module.exports = {
+	options,
+	handler,
+};
+
+const TAB = '\t';
+const NEWLINE = '\n';
+const FILE_HEADER = `<?php
+/* THIS IS A GENERATED FILE. DO NOT EDIT DIRECTLY. */
+$generated_i18n_strings = array(
+`;
+const FILE_FOOTER = `
+);
+/* THIS IS THE END OF THE GENERATED FILE */
+`;
+
+/**
+ * Parses module header translation strings.
+ *
+ * @param {WPTranslationsSettings} settings Translations settings.
+ *
+ * @return {[]string} List of translation strings.
+ */
+async function getTranslations( settings ) {
+	const moduleFilePattern = path.join( settings.directory, '*/load.php' );
+	const moduleFiles = await glob( path.resolve( '.', moduleFilePattern ) );
+
+	const moduleTranslations = await Promise.all(
+		moduleFiles.map( async ( moduleFile ) => {
+			const fileStream = fs.createReadStream( moduleFile );
+
+			const rl = readline.createInterface( {
+				input: fileStream,
+			} );
+
+			const headers = [ 'Module Name', 'Description' ];
+			const translationStrings = [];
+
+			for await ( const line of rl ) {
```

Comment:
@felixarntz why do we need to use `readline` and `for await`? Why not just `const file = fs.readFileSync( moduleFile, ... )` and then `const match = regex.exec( file );`? This will let us avoid using promises.

Reply:
I notably simplified this in https://github.com/WordPress/performance/pull/60/commits/430e98f5c4345769ac74e2794ea7d53232430b51, to not use promises and also to use just a single regex for all module headers.

I've also added a `filter` call to ensure that if a file doesn't have any matches it doesn't cause an error later.